### DISCUSSION TITLE
Does this plugin with work with Neovim nightly?

### DISCUSSION THREAD

**Discussion by nmsobri** [rmagatti/auto-session#212] [Q&A] [answered]
As title, is it? cause when i test with nightly build, it simply doesn't work

**Comment by rmagatti** [comment]
Yes it does. I'll need some more information to better understand what's happening

**Comment by nmsobri** [comment]
@rmagatti are you sure? cause when im using your plugin with nightly build, it doesn't even create any session file..but it work just fine on v0.8

**Reply by rmagatti** [reply]
Yep, pretty sure. What other information can you give me?

https://user-images.githubusercontent.com/2881382/227833419-a74cf295-b78b-4652-a987-51f57a7d0dda.mov

**Reply by nmsobri** [reply]
im sorry, im quite new to neovim.. I just open neovim and then exit it.. im expecting neovim to create session file inside `nvim-data/session` folder.. but nothing is created

**Reply by rmagatti** [reply]
Please record a video like the above or give a detailed explanation of exactly what you're doing. How are you opening Neovim? The readme of this plugin specifies behaviours when opening nvim in different ways, it would probably be a good idea to read through those as well.

**Reply by nmsobri** [reply]
In neovim v0.8 closing the terminal directly without closing nvim first ( ala using `:q` or `:qa` ) will save the session.. But in neovim v0.9, closing the terminal directly wont save the session

**Reply by rmagatti** [reply]
That's not something auto-session can control. It relies on the `VimExit` event Neovim publishes, the fact that a previous version of Neovim sent that event on terminal close is already surprising to me. I would have expected that flow to never have worked.

**Reply by nmsobri** [reply]
@rmagatti did you mean `VimLeave` ? IMHO, neovim should publish that event when its close.

**Reply by rmagatti** [reply]
Yes. Sorry, `VimLeave`. Like I mentioned, this isn't something auto-session has control over. If `VimLeave` gets triggered, session gets saved, if it doesn't (and there are cases it wouldn't) no session is saved.

Here's the `:h VimLeave` help.
```
 							*VimLeave*
VimLeave			Before exiting Vim, just after writing the
				.shada file.  Executed only once, like
				VimLeavePre.
				Use |v:dying| to detect an abnormal exit.
				Use |v:exiting| to get the exit code.
				Not triggered if |v:dying| is 2 or more.
```

**Reply by nmsobri** [reply, marked as answer]
I have open an issue on neovim.. https://github.com/neovim/neovim/issues/22972#issuecomment-1501073386 and one of Neovim contributor already make a PR to resolve this issue
